src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -14346,6 +14348,60 @@ instruct signumF_reg(vRegF dst, vRegF src, vRegF zero, vRegF one) %{
   ins_pipe(fp_uop_d);
 %}
 
+instruct onspinwait() %{
+  match(OnSpinWait);
+  ins_cost(INSN_COST);
+
+  format %{
+    $$template
+    switch (VM_Version::pause_impl_desc().inst()) {
+      case NOP:
+        $$emit$$"nop\t! membar_onspinwait\n"
+        for (unsigned int i = 1; i < VM_Version::pause_impl_desc().inst_count(); ++i) {
+          $$emit$$"\tnop\n"
+        }
+        break;
+      case ISB:
+        $$emit$$"isb\t! membar_onspinwait\n"
+        for (unsigned int i = 1; i < VM_Version::pause_impl_desc().inst_count(); ++i) {
+          $$emit$$"\tisb\n"
+        }
```

Comment:
The code to generate n copies of a pause_impl instruction would be much happier in the MacroAssembler.

Reply:
So, lose all the code here in C2 and push it down into a single macro.

Reply:
Done